pkg/controller/install/webhook.go

```diff
@@ -162,6 +182,71 @@ func (i *StrategyDeploymentInstaller) createOrUpdateValidatingWebhook(ogNamespac
 			log.Warnf("could not update ValidatingWebhookConfiguration %s", webhook.GetName())
 			return err
 		}
+
+		if len(desc.ConversionCrds) != 0 {
+			if err := createOrUpdateConversionCrds(desc, webhook.Webhooks[0].ClientConfig, i); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func createOrUpdateConversionCrds(desc v1alpha1.WebhookDescription, clientConfig admissionregistrationv1.WebhookClientConfig, i *StrategyDeploymentInstaller) error {
+
+	for index, ConversionCrd := range desc.ConversionCrds {
+		crd, err := i.strategyClient.GetOpLister().APIExtensionsV1().CustomResourceDefinitionLister().Get(desc.ConversionCrds[index])
+
+		// check if CRD is present on cluster
+		if err != nil {
+			log.Info("CRD not found %s, error: %s", desc.ConversionCrds[index], err.Error())
+		}
+
+		log.Info("Found CRD %s", desc.ConversionCrds[index])
+
+		// check if the CSV owns this CRD
+		foundCRD := false
+		csv, err := i.strategyClient.GetOpLister().OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(i.owner.GetNamespace()).Get(i.owner.GetName())
+		ownedCRDs := csv.Spec.CustomResourceDefinitions.Owned
+		for _, ownedCRD := range ownedCRDs {
+			if ownedCRD.Name == crd.GetName() {
+				log.Info("CSV %q owns CRD %q", csv.GetName(), crd.GetName())
+				foundCRD = true
+			}
+		}
+
+		// crd.Spec.Conversion.Strategy specifies how custom resources are converted between versions.
+		// Allowed values are:
+		// 	- None: The converter only change the apiVersion and would not touch any other field in the custom resource.
+		// 	- Webhook: API Server will call to an external webhook to do the conversion. This requires crd.Spec.preserveUnknownFields to be false.
+		// References:
+		//  - https://docs.openshift.com/container-platform/4.5/rest_api/extension_apis/customresourcedefinition-apiextensions-k8s-io-v1.html
+		// 	- https://kubernetes.io/blog/2019/06/20/crd-structural-schema/#pruning-don-t-preserve-unknown-fields
+		if (crd.Spec.Conversion.Strategy == "Webhook" || crd.Spec.Conversion.Strategy == "None") && foundCRD {
```

Comment:
could we handle the `Webhook` strategy and the `None` strategy clauses in entirely separate `if` blocks to simplify this logic? 

Looks like if its `Webhook` then we check preserve unknown fields (should we error out if `PreserveUnknownFields` is true?) and then update. 

If its `None` then we simply update. 

I guess this approach adds more lines of code but is more readable to someone looking at it for the first time IMO

Reply:
I believe we only need to update the webhookClientConfig when Strategy is Webhook. I updated the code accordingly and simplified it.